xray/backends/pynio_.py

```diff
@@ -0,0 +1,47 @@
+import numpy as np
+
+from .. import Variable
+from ..core.utils import FrozenOrderedDict, Frozen, NDArrayMixin
+from ..core import indexing
+
+from .common import AbstractDataStore
+
+
+class NioArrayWrapper(NDArrayMixin):
+    def __init__(self, array, ds):
+        self.array = array
+        self._ds = ds  # make an explicit reference because pynio uses weakrefs
```

Comment:
The weakrefs were used reluctantly after a period of great difficulty trying to get Python to clean up properly after closing files or deleting the file reference variable. This is an issue of the best demarcation between the C-API code and the Python code. It's possible that the code could be revised to make the group references in the Python code only. I just had to stop working on it for awhile because it was making me crazy and other things needed to be accomplished.